### ISSUE TITLE
Appium: 1.6.3: Android: fullReset does not grant permissions as fastReset

### ISSUE THREAD

**Issue by amedvedjev** [appium/appium#7451]
## The problem
When i start with "fastReset" appium grants permissions after app clear.
But when i start "fullReset" appium does uninstall app and install it with REINSTALL flag which cause in code do not grant permissions.

[debug] [ADB] Running '/Users/Aleksei/Library/Android/sdk/platform-tools/adb' with args: ["-P",5037,"-s","LGH8155e9892e6","shell","am","force-stop","XXXXXX"]
[debug] [ADB] Running '/Users/Aleksei/Library/Android/sdk/platform-tools/adb' with args: ["-P",5037,"-s","LGH8155e9892e6","uninstall","XXXXXX"]
[ADB] App was not uninstalled, maybe it wasn't on device?
[debug] [ADB] Getting connected devices...
[debug] [ADB] 1 device(s) connected
[debug] [ADB] Running '/Users/Aleksei/Library/Android/sdk/platform-tools/adb' with args: ["-P",5037,"-s","LGH8155e9892e6","shell","pm","install","-r","/data/local/tmp/0ed7d34cd848b1c70613725b6d925199.apk"]

**my code:**
```
                if (devicePlatform.contains("fullReset")) { // reinstall client
                    capabilities.setCapability(MobileCapabilityType.FULL_RESET, true);
                    capabilities.setCapability(MobileCapabilityType.NO_RESET, false);
                } else if (devicePlatform.contains("fastReset")) { // clears cache without reinstall
                    capabilities.setCapability(MobileCapabilityType.FULL_RESET, false);
                    capabilities.setCapability(MobileCapabilityType.NO_RESET, false);
                } else { // fast start
                    capabilities.setCapability(MobileCapabilityType.FULL_RESET, false);
                    capabilities.setCapability(MobileCapabilityType.NO_RESET, true);
                }

```

**appium-adb**
```
apkUtilsMethods.install = async function (apk, replace = true, timeout = 60000) {
  if (replace) {
    await this.adbExec(['install', '-r', apk], {timeout});
  } else {
    try {
      await this.adbExec(['install', apk], {timeout});
    } catch (err) {
      // on some systems this will throw an error if the app already
      // exists
      if (err.message.indexOf('INSTALL_FAILED_ALREADY_EXISTS') === -1) {
        throw err;
      }
      log.debug(`Application '${apk}' already installed. Continuing.`);
    }
  }
  if (apk.includes('.apk')) {
    let apiLevel = await this.getApiLevel();
    let targetSdk = null;
    try {
      targetSdk = await this.targetSdkVersionFromManifest(apk);
    } catch (e) {
      //avoiding logging error stack, as calling library function would have logged
      log.warn(`Ran into problem getting target SDK version; ignoring...`);
    }
    if (apiLevel >= 23 && (!targetSdk || targetSdk >= 23)) {
      await this.grantAllPermissions(await this.getPackageName(apk));
    }
  }
};
```

**Comment by jlipps** [comment]
@amedvedjev the flow in the code will grant all permissions regardless of whether we are going to replace or not. Can you try with appium 1.6.3?

**Comment by amedvedjev** [comment]
FIXED with latest 1.6.3. thanks.
